hibernate-core/src/main/java/org/hibernate/dialect/DB2Dialect.java

```diff
@@ -144,9 +144,9 @@ protected String columnType(int sqlTypeCode) {
 				// Note that 31 is the maximum precision DB2 supports
 				return columnType( DECIMAL );
 			case BLOB:
-				return "blob($l)";
+				return "blob";
 			case CLOB:
-				return "clob($l)";
+				return "clob";
```

Comment:
@beikov Why did you make this change.

The commit message doesn't explain why you did this.

Please at least separate it into its own commit.

Reply:
`DB2Dialect`, `DerbyDialect` and `MimerSQLDialect` were the only dialects that specified a length parameter, even though I'm pretty sure other dialects also allow specifying a length. In the end, the length is not really helpful anyway IMO, but the real problem is that we don't have a default length or a way to determine the LOB max size and we ended up with an unresolved variable in a test that I added. So I removed that to be consistent with the other dialects and fix the test I added.

Reply:
> but the real problem is that we don't have a default length or a way to determine the LOB max size and we ended up with an unresolved variable in a test that I added. 

OK, that's a big surprise to me, and I don't know how the existing tests didn't pick it up. But it seems like we should fix _that_ problem, not remove the ability to specify the length of a `blob` in DB2.

Because with the change you've made, if I'm not mistaken, it seems that any length the user explicitly specifies via the `@Length` annotation will now be ignored!

> `DB2Dialect`, `DerbyDialect` and `MimerSQLDialect` were the only dialects that specified a length parameter

Sure, this is all very platform-specific stuff. But if the platform _supports_ a length parameter then we shouldn't just ignore it.

So this looks like the wrong "fix" for the problem you encountered.

Reply:
>OK, that's a big surprise to me, and I don't know how the existing tests didn't pick it up. But it seems like we should fix that problem, not remove the ability to specify the length of a blob in DB2.
>Because with the change you've made, if I'm not mistaken, it seems that any length the user explicitly specifies via the @Column annotation will now be ignored!

How about we support length in a unified way across all databases in a follow up fix? We could generate a check constraint to assert the length limit if the type doesn't support specifying capacity.

Reply:
Well I think some databases (Postgres, if I'm not mistaken) simply don't support a maximum length. But  I don't recall any details or this.

Reply:
For MySQL it surely already works correctly.

Reply:
>Well I think some databases (Postgres, if I'm not mistaken) simply don't support a maximum length. But I don't recall any details or this.

You are right, but I think these databases usually support check constraints, so we could enforce a length here if we think it is reasonable to do so.

>For MySQL it surely already works correctly.

Not sure what you mean. AFAICS MySQL has no possibility to specify the length for LOB types.

Reply:
Let's continue that discussion on this issue I just created: https://hibernate.atlassian.net/browse/HHH-15155

Reply:
> Not sure what you mean. AFAICS MySQL has no possibility to specify the length for LOB types.

It has `blob`, `mediumblob`, `longblob`, etc. I'm sure we already pick the right one based on the specified `length`.

Reply:
Right, but there is no length variable in the DDL type "name"/pattern that can specify exact capacity.